examples/albef/model.py

```diff
@@ -448,5 +627,35 @@ def albef_model_for_vqa(config: dict, pretrained: bool = False) -> ALBEFModelFor
     model = ALBEFModelForVQA(albef_model, decoder, loss)
 
     if pretrained:
-        load_module_from_url(model, _ALBEF_PRETRAINED_URLS["vqa"])
+        checkpoint = torch.hub.load_state_dict_from_url(
+            _ALBEF_PRETRAINED_URLS["vqa"], map_location="cpu"
+        )
+        model.load_state_dict(checkpoint)
+    return model
+
+
+def albef_model_for_retrieval(
```

Comment:
I think we have been leaning more towards using explicit args in the builder functions instead of configs. Not a huge deal, but something worth thinking about for consistency (can always add a to-do to unify these)

Reply:
I'm on the fence with this, the closest comparison is [flava_model](https://github.com/facebookresearch/multimodal/blob/e73ad6db1d2f93b55a4290649470e7ce005dbb2b/torchmultimodal/models/flava/model.py#L418) which flattens each encoder's args, leading to a gigantic function signature. I would lean towards being consistent with the codebase, but I see the appeal of having just one dict with all the args.

another thing I would prioritize is the ability to just call this function and have it create everything with default parameters. at this high level you want to make it as easy as possible for the user. flattening the args let's you do that. another option is to break it down into kwargs for each encoder. so the function will have `vision_encoder_args`, `text_encoder_args`, etc as params that are optional. that way you can still take advantage of using default values if the user does not want to specify anything.